### PR TITLE
Add opam 2.2

### PR DESCRIPTION
**Pull request by mtelvers**

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
This already exists. See line 216

**Comment by mtelvers** [comment]
@kit-ty-kate Should I now separate opam 2.2 and opam master for testing against the two as per https://github.com/ocurrent/opam-repo-ci/issues/331?

**Comment by kit-ty-kate** [comment]
sure

File: src-opam/opam.ml [hunk 1]
```diff
@@ -212,12 +213,13 @@ let opam_master_branch opam_master_hash =
     enable_0install_solver = true;
     with_vendored_deps = true;
     public_name = "opam-dev";
-    aliases = [ "opam-2.2" ];
-    (* TODO: Remove/update when opam 2.2 is branched *)
+    aliases = [ "" ];
```
Comment (kit-ty-kate) [inline src-opam/opam.ml:216]:
```suggestion
    aliases = [];
```

File: src-opam/opam.ml [hunk 2]
```diff
@@ -293,22 +304,23 @@ let copy_opams ~src ~dst opam_branches =
...
+let copy_opams_windows opam_branches =
+  List.fold_left
+    (fun acc { branch; public_name; aliases; _ } ->
+      acc
+      (* Docker doesn't allow copying executables to /usr/local/bin *)
+      @@ copy ~from:"opam-builder"
+           ~src:[ {|c:\cygwin64\usr\local\bin\opam-|} ^ branch ^ ".exe" ]
+           ~dst:({|c:\|} ^ public_name ^ ".exe")
+           ()
+      @@ run {|move c:\%s.exe c:\cygwin64\usr\local\bin|} public_name
```
Comment (MisterDA) [inline src-opam/opam.ml:316]:
```suggestion
           ~src:[ {|C:\cygwin64\usr\local\bin\opam-|} ^ branch ^ ".exe" ]
           ~dst:({|C:\|} ^ public_name ^ ".exe")
           ()
      @@ run {|move C:\%s.exe c:\cygwin64\usr\local\bin|} public_name
```

**Review by MisterDA — COMMENTED** [review]
(nitpicking)
That's really cool work, thanks Mark! Eager to see this in ocaml-ci!

File: src-opam/opam.ml [hunk 3]
```diff
@@ -237,12 +236,33 @@ let create_opam_branches opam_hashes =
...
+  ( opam_master_hash,
+    [
+      {
+        branch = "2.2";
+        hash = opam_2_2_hash;
+        enable_0install_solver = true;
+        with_vendored_deps = true;
+        public_name = "opam-2.2";
+        aliases = [];
+      };
```
Comment (MisterDA) [inline src-opam/opam.ml:263]: Maybe this could be factored with the other definition above?

Comment (mtelvers) [reply src-opam/opam.ml]: I've updated the code, so it is defined once and used a filter to remove the branches we don't want to build under Windows.

**Review by kit-ty-kate — APPROVED** [review]

File: src-opam/opam.ml [hunk 4]
```diff
@@ -237,12 +236,23 @@ let create_opam_branches opam_hashes =
...
       opam_master_branch opam_master_hash;
     ] )
 
 let create_opam_branches_windows opam_hashes =
-  let { opam_master_hash; _ } = opam_hashes in
-  (opam_master_hash, [ opam_master_branch opam_master_hash ])
+  let opam_master_hash, branches = create_opam_branches opam_hashes in
+  ( opam_master_hash,
+    List.filter
+      (fun { branch; _ } -> branch = "master" || branch = "2.2")
```
Comment (kit-ty-kate) [inline src-opam/opam.ml:254]:
```suggestion
      (fun { branch; _ } -> branch >= "2.2")
```

Comment (kit-ty-kate) [reply src-opam/opam.ml:254]: more future proof

File: src-opam/opam.ml [hunk 5]
```diff
@@ -212,12 +210,13 @@ let opam_master_branch opam_master_hash =
     enable_0install_solver = true;
     with_vendored_deps = true;
     public_name = "opam-dev";
-    aliases = [ "opam-2.2" ];
-    (* TODO: Remove/update when opam 2.2 is branched *)
+    aliases = [];
```
Comment (shonfeder) [inline src-opam/opam.ml:213]: Assuming we still want to keep this field around even tho the todo is resolved, would it make sense, and help in the future, to add a comment on the `alias` field in the type definition (line 203) to note what this is meant to be used for?

**Review by shonfeder — APPROVED** [review]

File: src-opam/opam.ml (hunk 5 above)
Comment (mtelvers) [reply src-opam/opam.ml:213]: `aliases` is still used to link 2.0 to the default.  See L228.  The to do here referred to aliasing the `master` branch to `2.2` even though 2.2 was not released.

File: src-opam/opam.ml (hunk 4 above)
Comment (mtelvers) [reply src-opam/opam.ml:254]: ok, I'll add a comment in the code as well

**Pull request merged**